### PR TITLE
Convert `assemble_nexus_url` to modern API

### PR DESCRIPTION
**Pull request by alexjfisher**

### CODE REVIEW

**Comment by alexjfisher** [comment]
This is the easy bit of https://github.com/voxpupuli/puppet-archive/pull/329
Easy in so much as this function doesn't do anything unsafe like requiring `puppet_x/bodeco/util` (which isn't an environment safe thing to do).

**Comment by alexjfisher** [comment]
> Due to the rename, is this a breaking change?

Depends on whether you think the original function was ever considered public.  I think probably not.

**Comment by ekohl** [comment]
Probably not

File: lib/puppet/functions/archive/assemble_nexus_url.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  # @param nexus_url
+  #   The base nexus URL
+  # @param params
+  #   The query parameters as a hash
+  #
+  # @return [Stdlib::HTTPUrl]
+  #   The assembled URL
+  dispatch :default_impl do
+    param 'Stdlib::HTTPUrl', :nexus_url
+    param 'Hash', :params
```
Comment (alexjfisher) [inline lib/puppet/functions/archive/assemble_nexus_url.rb]: I should probably put the return type in here too.

**Pull request merged**